crates/data-generator/src/blocks/output.rs

```diff
@@ -0,0 +1,586 @@
+    use quote::{quote, ToTokens};
+
+    use crate::{blocks::model::PropertyVariant, name::Name};
+
+    #[derive(Debug)]
+    pub struct PropertyDefault<'a> {
+        pub enum_name: &'a Ident, // WeepingVines
+        pub def_var_name: Ident,
+    }
+
+    impl<'a> PropertyDefault<'a> {
+        pub fn new(name: &'a Name, value: &PropertyVariant) -> Self {
+            PropertyDefault {
+                enum_name: name.pascal_case(),
+                def_var_name: value.variant().into_pascal_case(),
+            }
+        }
+    }
+
+    impl ToTokens for PropertyDefault<'_> {
+        fn to_tokens(&self, tokens: &mut TokenStream) {
+            let enum_name = &self.enum_name;
+            let def_var_name = &self.def_var_name;
+
+            tokens.extend(quote! {
+                impl Default for #enum_name {
+                    fn default() -> Self {
+                        Self::#def_var_name
+                    }
+                }
+            });
+        }
+    }
+}
+
+mod prop_convert {
+    use proc_macro2::{Ident, Literal, TokenStream};
+    use quote::{quote, ToTokens};
+
+    use crate::{
+        blocks::model::{PropertyVariant, PropertyVariants},
+        name::Name,
+    };
+
+    pub struct PropertyConvert<'a> {
+        pub prop_name: &'a Ident, // HoneyLevel
+        pub prop_vars: &'a PropertyVariants,
+    }
+
+    impl<'a> PropertyConvert<'a> {
+        pub fn new(name: &'a Name, variants: &'a PropertyVariants) -> Self {
+            Self {
+                prop_name: name.pascal_case(),
+                prop_vars: variants,
+            }
+        }
+    }
+
+    impl ToTokens for PropertyConvert<'_> {
+        fn to_tokens(&self, tokens: &mut TokenStream) {
+            match &self.prop_vars {
+                PropertyVariants::Bool => {
+                    PropertyNotBool(&self.prop_name).to_tokens(tokens);
+                    PropertyFromBool(&self.prop_name).to_tokens(tokens);
+                    PropertyIntoBool(&self.prop_name).to_tokens(tokens);
+                }
+                PropertyVariants::Numeric(numbers) => {
+                    let impl_from_u8 = PropertyFromU8 {
+                        enum_name: &self.prop_name,
+                        numbers: &numbers,
+                    };
+                    impl_from_u8.to_tokens(tokens);
+                }
+                _ => {}
+            }
+        }
+    }
+
+    pub struct PropertyFromU8<'a> {
+        pub enum_name: &'a Ident,
+        pub numbers: &'a Vec<u8>,
+    }
+
+    impl ToTokens for PropertyFromU8<'_> {
+        fn to_tokens(&self, tokens: &mut TokenStream) {
+            let mut match_arms = TokenStream::new();
+            for &num in self.numbers.iter() {
+                let u8_num = Literal::u8_unsuffixed(num); // "4"
+                let roman_num = PropertyVariant::Numeric(num).variant();
+                let roman_num = roman_num.into_pascal_case(); // "IV"
+
+                // "4 => Self::IV,"
```

Comment:
```suggestion
                let roman_num = roman_num.into_pascal_case(); // "Iv"

                // "4 => Self::Iv,"
```
Because this is pascal case

Reply:
For me, this is an unequivocal "no". I see where you're coming from, and I agree that we should generally adhere to Rust's standards, but we're talking Roman numerals here, not an abbreviation or whatnot. This is uncultured, indecent, and generally a slap in the face to the Romans. @JarvisCraft

Reply:
These are just names that we need to substitute numbers; who cares how they're formatted?